core/src/test/java/io/substrait/type/proto/ExtensionRoundtripTest.java

```diff
@@ -174,6 +175,21 @@ void join() {
     verifyRoundTrip(rel);
   }
 
+  @Test
+  void hashJoin() {
+    int[] left_keys = {};
```

Comment:
Should we have some tests with non-empty keys to make sure that's working correctly?

Reply:
+1 to using non-empty keys.

Reply:
@vbarua @EpsilonPrime 
Looking into the `commonTable` initialization, we do have empty lists to create it, so if we use non-empty lists for keys, we may need to introduce a commonTable with non-empty initialization? Am I following this incorrectly?


```java
final Rel commonTable =
      b.namedScan(Collections.emptyList(), Collections.emptyList(), Collections.emptyList());
```

Reply:
> we may need to introduce a commonTable with non-empty initialization? 

This is correct. You can use the `b.namedScan` method to create a table with columns, something like
```java
b.namedScan(List.of("example"), List.of("a"), List.of(R.STRING)
```

One small structural thing. The tests in `ExtensionRoundtripTest.java` verify that _extensions_ roundtrip. The test that you've added already does this perfectly well.

For adding a test using non-empty keys, could I ask that you introduce something like `JoinRountripTest.java` (in a similar vein to `AggregateRoundtripTest.java`) and do it there. That will also be a good place to add nested join tests later.

Reply:
@vbarua I will defintely add a test class as you have susggested here. 👍